### PR TITLE
Use ast parsing & provide flake8 entry point

### LINKED ISSUE #4: Writing a flake8 plugin for docargs
Flake8 has support for [writing plugins](http://flake8.pycqa.org/en/latest/plugin-development/index.html), which allow it to be extended with additional checks. It would be useful if we could write a plugin for docargs: the main benefit over using docargs as a standalone CLI is that we'd benefit from existing flake8 integrations for text editors and IDEs, so docargs warnings would show inline in the same way as other warnings.

### PR DESCRIPTION
**Pull request by janfreyberg**
This is an implementation that parses the abstract syntax tree instead of importing code. This now means it works on arbitrary code.

This also closes #4 by providing a flake8 entry point for docargs.

@srstevenson this is now on PyPI as pre-release - could you give it a quick spin and let me know any immediate issues?

### CODE REVIEW

File: docargs/__init__.py [hunk 1]
```diff
@@ -1,180 +1,69 @@
...
+                )
+
     if failed:
         sys.exit(1)
     else:
+        click.echo(Fore.GREEN + "All arguments are documented ✓")
         sys.exit(0)
 
 
+def cli_error(file_name, statement, underocumented, overdocumented):
```
Comment (42B) [inline docargs/__init__.py]: "underocumented" -> "underdocumented"?

File: docargs/flake8.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.tree = tree
+        self.filename = filename
+
+    def run(self):
+        tree = self.tree
+        if self.filename == "stdin":
+            lines = stdin_utils.stdin_get_value()
+            tree = ast.parse(lines)
+
+        for statement, underocumented, overdocumented in check(tree):
```
Comment (42B) [inline docargs/flake8.py]: "underocumented" -> "underdocumented"?

File: docargs/__init__.py (hunk 1 above)
Comment (janfreyberg) [reply docargs/__init__.py]: yes, thanks!

**Comment by srstevenson** [comment]
@janfreyberg I installed the release from PyPI and gave it a quick test, and it worked well! I'll endeavour to test it more thoroughly in the coming week.

**Pull request merged**